inc/class-wpseo-utils.php

```diff
@@ -524,6 +526,29 @@ public static function clear_sitemap_cache( $types = array() ) {
 	}
 
 	/**
+	 * Clear sitemap transient caches
+	 *
+	 * @param array $types Set of sitemap types to clear.
+	 */
+	public static function clear_sitemap_transient_cache( $types = array() ) {
+		// Always delete the main index sitemaps cache, as that's always invalidated by any other change.
+		if ( ! in_array( 1, $types, true ) ) {
+			array_unshift( $types, 1 );
+		}
+
+		foreach ( $types as $type ) {
+			$page = 0;
+			do {
+				$key   = sprintf( 'wpseo_sitemap_cache_%s_%d', $type, ++ $page );
```

Comment:
`$page` will never be 0 here due to the incrementing taking place before evaluation of the variable.
It would be far more explicit if `$page = 1;` on line 540 and incrementing later on in the code.
Using `$page += 1;` when incrementing would be even more explicit.